### DISCUSSION TITLE
URL path with colon (":") results in 404 not found on POST?

### DISCUSSION THREAD

**Discussion by dwinston** [fastapi/fastapi#8513] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.post("/items:queue")
def queue_new_item(item: dict):
    return "An item has been queued for later inclusion."
```

### Description

* Open the browser and `POST /items:queue`
    * For example, go to http://127.0.0.1:8000/docs#/default/queue_new_item_items_queue_post
    * Click to Try it out
    * Click to Execute (default request body of `{}` is fine)
 * I expect the string response "An item has been queued for later inclusion."
 * I get a 404 not found response, with JSON body {"detail": "Not Found"}

### Operating System

macOS

### FastAPI Version

0.78.0

### Python Version

Python 3.9.12

**Comment by dwinston** [comment]
I realize this may be an issue upstream with Starlette, so I have started a discussion topic there: <https://github.com/encode/starlette/discussions/1673>.

**Comment by Kludex** [comment, marked as answer]
This is solved on Starlette 0.20.2.

FastAPI still doesn't support that Starlette version.

Note: By the time this was written, FastAPI is on version 0.78.0.

**Comment by yuanjie-ai** [comment]
![Uploading image.png…]()

**Comment by bodograumann** [comment]
Duplicate of #4892
